task6/Working with posts.js

```diff
@@ -0,0 +1,251 @@
+new post("24", "Ava.jpg", "incredible4",  new Date('2020-12-07T10:14:22'), "Hello111"),
+new post("25", "Ava.jpg", "incredible4",  new Date('2020-12-15T02:38:40'), "Hello481"),
+new post("26", "Ava.jpg", "incredible4",  new Date('2020-01-05T17:41:38'), "Hello211"),
+new post("27", "Ava.jpg", "incredible4",  new Date('2020-06-07T13:53:03'), "Hello154", ["#hello", "#hai"]),
+new post("28", "Ava.jpg", "incredible4",  new Date('2020-01-23T12:15:11'), "Hello564"),
+new post("29", "Ava.jpg", "incredible4",  new Date('2020-07-05T20:30:55'), "Hello931"),
+new post("30", "Ava.jpg", "incredible4",  new Date('2020-04-25T00:08:16'), "Hello990"),
+new post("31", "Ava.jpg", "incredible5",  new Date('2020-07-10T09:20:03'), "Hello633"),
+new post("32", "Ava.jpg", "incredible5",  new Date('2020-09-27T18:55:54'), "Hello468"),
+new post("33", "Ava.jpg", "incredible5",  new Date('2020-12-27T12:39:01'), "Hello530", ["#hai"]),
+new post("34", "Ava.jpg", "incredible5",  new Date('2020-12-20T18:54:23'), "Hello927", ["#ni hao"]),
+new post("35", "Ava.jpg", "incredible5",  new Date('2020-07-19T11:25:13'), "Hello549"),
+new post("36", "Ava.jpg", "incredible5",  new Date('2020-10-26T40:03:24'), "Hello616"),
+new post("37", "Ava.jpg", "incredible5",  new Date('2020-12-23T17:20:30'), "Hello682"),
+new post("38", "Ava.jpg", "incredible5",  new Date('2020-07-28T00:46:26'), "Hello208"),
+new post("39", "Ava.jpg", "incredible5",  new Date('2020-10-22T02:59:26'), "Hello940"),
+new post("40", "Ava.jpg", "incredible5",  new Date('2020-03-15T21:40:13'), "Hello999")
+];
+
+
+class editInfo{
+	constructor(description, hashTags = [], photoLink="") {
+		this.description = description;
+		this.hashTags = hashTags;
+		this.photoLink = photoLink;
+	}
+}
+
+class filterParams {
+	constructor (author, date, hashTags=[]) {
+		this.author = author;
+		this.date = date;
+		this.hashTags = hashTags;
+	}
+}
+
+function inOneDay(date1, date2) {
+	return date1.getFullYear() == date2.getFullYear()
+		 && date1.getMonth() == date2.getMonth()
+		 && date1.getDate() == date2.getDate();
+}
+
+function hasAnyTag(hashTagsOfPost, hashTags) {
+	return hashTagsOfPost.some(hashTagOfPost=>(hashTags.some(hashTag=>hashTag==hashTagOfPost)));
+}
+
+function getPostsByFilter(filterConfig) {
+	return posts.filter(element=>
+		(element.author == filterConfig.author) &&
+		((filterConfig.date == null) || inOneDay(element.createdAt, filterConfig.date)) &&
+		 ((filterConfig.hashTags == []) || hasAnyTag(element.hashTags, filterConfig.hashTags))
+		 );
+}
+
+function compDates(x, y) {
+	return x.createdAt < y.createdAt ? -1 : x.createdAt > y.createdAt ? 1 : 0;
+}
+
+function getPosts(filterConfig, skip=0, top=10) {
+	if (skip < 0 || top <= 0) {
+		return [];
+	}
+	let postsByFilter = getPostsByFilter(filterConfig).sort(compDates);
+	return postsByFilter.slice(skip, skip + top);
+}
+
+function getPost(ID) {
+	return posts.find(function(item, index, array) {
```

Comment:
Plz, use arrow function as a callback. Also, index and array params are not used, so just remove them.